### DISCUSSION TITLE
HELP: Dropdown with VirtualScroll

### DISCUSSION THREAD

**Discussion by mrc-bsllt** [primefaces/community#1531] [PrimeVue]
Hi everyone, I need to ask you for help, I am working on a project where I use nuxt-primevue. I am trying to use a dropdown with the virtualScroll functionality in lazyLoad, but I can't achieve the desired result, i.e., I have an initial array of 10 items, and when I get to the last item, I would like to make an API call, fetch another 10 items and add them to the initial array, and so on. I can't figure out through the documentation how to control this, now I have a reproduction on stackblitz. Can anyone help me? Thanks!
reproduction: https://stackblitz.com/edit/nuxt-starter-rnzinb?file=app.vue

**Comment by ipimpat** [comment]
I don't think virtual scroll was designed to implement "infinite scroll".

From my understanding, with virtual scrolling, you need to provide an array that represents the total dataset size. This array doesn't initially contain actual data but serves as a placeholder to manage which data is loaded based on the user's scroll position. The virtual scroll component then determines which data to dynamically load through a callback.

In contrast, "infinite scroll" fetches new data from an API, appending it page by page as the user scrolls down, thus dynamically expanding the dataset's size, which could be easily implemented using something similar to this: https://vueuse.org/core/useInfiniteScroll and appears more straightforward way of doing it.

I'm still not quite sure about the practical applications of virtual scrolling in PrimeVue, as it seems to be designed with very specific use cases in mind.

I would appreciate insights from someone within the PrimeVue community on this matter, as it seems to be a recurring point of discussion.

Also given that PrimeVue's component lacks slots that allow you to override internal logic, I don't think you can do much to alter its behavior either.
